metatheory/metatheory.cabal

```diff
@@ -0,0 +1,1037 @@
+cabal-version:       2.4
+name:                plc-agda
```

Comment:
Can we keep the package name the same as the cabal file name? Otherwise it's very confusing. It's bad enough that the directory name doesn't match.

Maybe we should rename both the directory and the package to `plutus-metatheory`, and have the executable remain `plc-agda`?